features/include/pcl/features/normal_3d.h

```diff
@@ -186,6 +186,49 @@ namespace pcl
     }
   }
 
+  /** \brief Flip (in place) normal to get the same sign of the mean of the normals specified by normal_indices.
+    * 
+    * The method is described in:
+    * A. Petrelli, L. Di Stefano, "A repeatable and efficient canonical reference for surface matching", 3DimPVT, 2012
+    * A. Petrelli, L. Di Stefano, "On the repeatability of the local reference frame for partial shape matching", 13th International Conference on Computer Vision (ICCV), 2011
+    * \param[in] normal_cloud input cloud of normals used to compute the mean
+    * \param[in] normal_indices indices of normals used to compute the mean 
+    * \param[in] normal input normal to flip. normal is modified by the function.
+    * \return false if normal_indices does not contain any valid normal.
+    * \ingroup features
+    */
+  template<typename PointNT> inline bool
+  flipNormalTowardsNormalsMean ( pcl::PointCloud<PointNT> const &normal_cloud,
+                                 std::vector<int> const &normal_indices,
+                                 Eigen::Vector3f &normal)
+  {
+    Eigen::Vector3f normal_mean = Eigen::Vector3f::Zero();
+
+    bool at_least_one_valid_point = false;
+    for (size_t i = 0; i < normal_indices.size (); ++i)
+    {
+      const PointNT& curPt = normal_cloud[normal_indices[i]];
+
+      if(pcl::isFinite(curPt))
+      {
+        normal_mean += curPt.getNormalVector3fMap ();
```

Comment:
Do we require that normals have unit length? If not then they will have different "weight".
If we require (and I assume we do), then would be nice to add this to the docstring. Further, there is no need for the `at_least_one_valid_point` variable. You can just sum up finite normals and check if `normal_mean` is not zero.